### PR TITLE
feat: My Collection analytics tracking

### PR DESCRIPTION
**Pull request by pepopowitz**
The type of this PR is: **Feature**

This PR partially addresses [CX-886]. The remaining tracking events will be handled in follow-up PRs.

### Description

Adds (some) analytics tracking for the My Collection app.

Also includes a few minor cleanup things (grammar, doc URLs, an analytics property [recently changed in cohesion](https://github.com/artsy/cohesion/pull/141/files/cb29308371f77d64fc0f5d09d7fbac6cd16c69fa#diff-3a0a8bde11a4618e1f97edf6854e5526110fb10396ffd9ec0e7100530e135554)). 

A collaboration with @annacarey!

I have not included screenshots because there would be so many. If you'd like me to attach some, please push back.

### Events included in this PR

Tap tracking for: 
- [x] Add artwork
- [x] Edit artwork
- [x] Delete artwork
- [x] TappedCollectedArtwork
- [x] TappedCollectedArtworkImages
- [x] TappedInfoBubble for 'Demand Index'
- [x] TappedInfoBubble for 'Price estimate'
- [x] TappedInfoBubble for 'Artist Market Statistics'
- [x] TappedInfoBubble for 'Auction results'

### Events that will be tracked in follow-up PRs

Screen tracking for: 
- Home (MyCollection.tsx) 
- Artwork (MyCollectionArtwork.tsx)

Tap tracking for:
- TappedSell
- TappedShowMore for ‘Learn more’
- TappedShowMore for ‘See all articles’
- TappedShowMore for ‘Explore auction results’
- TappedShowMore for ‘View more’

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

[CX-434]: https://artsyproduct.atlassian.net/browse/CX-434
[CX-886]: https://artsyproduct.atlassian.net/browse/CX-886

### CODE REVIEW

File: src/lib/Scenes/MyCollection/Screens/Artwork/Components/ArtworkInsights/MyCollectionArtworkArtistAuctionResults.tsx [hunk 1]
```diff
@@ -130,3 +136,15 @@ export const MyCollectionArtworkArtistAuctionResultsFragmentContainer = createFr
     `,
   }
 )
+
+const tracks = {
+  tappedInfoBubble: (internalID: string, slug: string) => {
```
Comment (damassi) [inline src/lib/Scenes/MyCollection/Screens/Artwork/Components/ArtworkInsights/MyCollectionArtworkArtistAuctionResults.tsx:141]: Ohh, I like this pattern a lot 💯

**Review by mikehrom — APPROVED** [review]
Everything looks good from my end, I would say the outstanding, highest priority tracks are for screens, which we do have a [schema](https://github.com/artsy/cohesion/pull/142) for in cohesion now.

I'm entertaining editing the schema to include a prop for zero state, which I didn't originally think was important, but I'm seeing a few more places where we might find it useful (inbox, for example).

**Comment by mikehrom** [comment]
TappedSell is also the most important unaddressed tap so far, too. Would be great to know about people entering the consign flow from this feature.

File: src/lib/Scenes/MyCollection/Screens/Artwork/Components/ArtworkInsights/MyCollectionArtworkArtistAuctionResults.tsx (hunk 1 above)
Comment (pepopowitz) [reply src/lib/Scenes/MyCollection/Screens/Artwork/Components/ArtworkInsights/MyCollectionArtworkArtistAuctionResults.tsx:141]: It's in the docs! https://github.com/artsy/eigen/blob/5c781950a98ba6846d1b8974a3c6356d58d87116/docs/analytics_and_tracking.md#L21

Comment (damassi) [reply src/lib/Scenes/MyCollection/Screens/Artwork/Components/ArtworkInsights/MyCollectionArtworkArtistAuctionResults.tsx:141]: Nice -- we should adopt this over in force, much better than cluttering render functions.

**Comment by pepopowitz** [comment]
@mikehrom (and others but especially Mike): I decided to cut this PR off before finishing _all_ My Collection tracking, as the volume of changes was becoming un-reviewable. I've updated the body of this PR with (hopefully) clear definition of which events are included here and which will come next.

File: docs/analytics_and_tracking.md [hunk 2]
```diff
@@ -3,20 +3,23 @@
...
+
 **instead of**:
-```typescript
 
+```typescript
 AppRegistry.registerComponent("Inquiry", () => Inquiry)
 ```
 
-This will make sure that `Inqury` is wrapped properly with the tracking context it's needed, regardless of it being a func or a class component.
+This will make sure that `Inquiry` is wrapped properly with the tracking context it's needed, regardless of it being a func or a class component.
```
Comment (pepopowitz) [inline docs/analytics_and_tracking.md:17]: This is the only significant change in this noisy file update.

File: src/lib/Scenes/BottomTabs/bottomTabsConfig.tsx [hunk 3]
```diff
@@ -16,7 +16,7 @@ export const bottomTabsConfig: {
   },
   inbox: {
     route: "/inbox",
-    analyticsDescription: "messages",
+    analyticsDescription: "inbox",
```
Comment (pepopowitz) [inline src/lib/Scenes/BottomTabs/bottomTabsConfig.tsx:19]: This was associated with a [cohesion change](https://github.com/artsy/cohesion/pull/141).

File: src/lib/Scenes/MyCollection/Screens/Artwork/Components/ArtworkInsights/InfoButton.tsx [hunk 4]
```diff
@@ -9,10 +9,11 @@ interface InfoButtonProps {
   title: string
   modalContent: JSX.Element
   modalTitle?: string
+  onPress?: () => void
```
Comment (pepopowitz) [inline src/lib/Scenes/MyCollection/Screens/Artwork/Components/ArtworkInsights/InfoButton.tsx:12]: Adding a hook here for each individual InfoButton parent to track its own analytics.

**Review by pepopowitz — COMMENTED** [review]
Added a few notes for review.

File: docs/preferred_practices.md [hunk 5]
```diff
@@ -115,7 +115,7 @@ _Most_ interactions are made through a "SwitchBoard" to open links. Other intera
 
 ### Analytics
 
-There is [extensive inline documentation](https://github.com/artsy/emission/blob/7a4eb922cee70c621c9938bfda1db460e65414fc/src/lib/utils/track.ts#L175-L359) in our tracking code, including examples.
+There is [extensive inline documentation](https://github.com/artsy/eigen/blob/3281cd9ba54b1d316b68375f46389974a22a0091/src/lib/utils/track/index.tsx#L10-L212) in our tracking code, including examples.
```
Comment (brainbicycle) [inline docs/preferred_practices.md:118]: Thanks for updating this 🙌

**Review by brainbicycle — APPROVED** [review]

File: src/lib/Scenes/MyCollection/Screens/Artwork/Components/ArtworkInsights/__tests__/MyCollectionArtworkArtistAuctionResults-tests.tsx [hunk 6]
```diff
@@ -96,4 +111,26 @@ describe("MyCollectionArtworkArtistAuctionResults", () => {
     wrapper.root.findAllByType(CaretButton)[0].props.onPress()
     expect(navigate).toHaveBeenCalledWith("/artist/artist-slug/auction-results")
   })
+
+  it("tracks analytics event when info button is tapped", () => {
```
Comment (brainbicycle) [inline src/lib/Scenes/MyCollection/Screens/Artwork/Components/ArtworkInsights/__tests__/MyCollectionArtworkArtistAuctionResults-tests.tsx:115]: 🎉

**Comment by brainbicycle** [comment]
Looks good to me 🙌 feel free to merge when ready, wasn't sure if you were waiting for any other reviews!

**Pull request merged**